src/translator/annotation.h

```diff
@@ -122,6 +122,9 @@ struct AnnotatedText {
   std::string text;       ///< Blob of string elements in annotation refers to.
   Annotation annotation;  ///< sentence and (sub-) word annotations.
 
+  /// Tag positions at char-level in the source text
+  std::vector<ByteRange> tagPositions;
```

Comment:
There is `using TagPositions = std::vector<ByteRange>;` in `definitions.h` file. You can just use it.